### ISSUE TITLE
Refactoring Notebook code

### ISSUE THREAD

**Issue by mohamed-yehia311** [mohamed-yehia311/med-reasoning-tuning#4]
in these days we need to finish fine-tuneing our model in the dataset of COT.

therefore, we need to refactor our notebook code and adding extra features:
1. use tensor flow instead of torch 
2. use Dynamic padding to optimize recourses usage
3. adjusting hyperparameters for these new dataset 
4. refining the prompt
5. fine tuning the model in the **three COT datasets** and see the model performance in the test set
